File: Chap1-Events/1-4EventPropagationDemo/parentlineedit.cpp
#include "parentlineedit.h"
#include <QDebug>
#include <QKeyEvent>

ParentLineEdit::ParentLineEdit(QWidget *parent) : QLineEdit(parent)
{

}

void ParentLineEdit::keyPressEvent(QKeyEvent *event)
{
    qDebug() << "ParentLineEdit keyPressEvent";
    qDebug()  <<" ParentLineEdit Accepted : " << event->isAccepted();
    QLineEdit::keyPressEvent(event);
}

File: Chap1-Events/1-4EventPropagationDemo/parentlineedit.py
from PySide6.QtWidgets import QLineEdit
from PySide6.QtGui import QKeyEvent

class ParentLineEdit(QLineEdit):
    def __init__(self, parent=None):
        super().__init__(parent)
    
    def keyPressEvent(self, event: QKeyEvent):
        print("ParentLineEdit keyPressEvent")
        print(f"ParentLineEdit Accepted: {event.isAccepted()}")
        super().keyPressEvent(event)